src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,161 @@
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via the `trace` envelope header.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers.
+While all of these values are optional, SDKs should make their best effort to add as many of them to the baggage header as possible when starting a trace.
+
+- `sentry-traceid` - The original trace ID as generated by the SDK
+- `sentry-publickey` - Public key as defined by the user via the DSN in the SDK options
+- `sentry-release` - The release as defined by the user in the SDK options
+- `sentry-environment` - The environment as defined by the user in the SDK options
+- `sentry-transaction` - The name of the trace's origin transaction in unparameterized (raw) format
+- `sentry-userid` - User ID as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-usersegment` - User segment as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-samplerate` - Sample rate as defined by the user in the SDK options
+
+SDKs must set all of the keys in the form of "`sentry-[name]`".
+The prefix "`sentry-`" acts to identify key-value pairs set by Sentry SDKs.
+Additionally, we chose `[name]` to be written in "snake case" without any underscore ( `_` ) characters. This naming convention is the most language agnostic.
+
+### Envelope Header
+
+Dynamic Sampling Context is transferred to Sentry through the `trace` envelope header.
+The value of this header corresponds directly to the definition of <Link to="/sdk/performance/trace-context/#trace-context">Trace Context</Link>.
+
+When a transaction is reported to Sentry, the Dynamic Sampling Context must be mapped to the `trace` envelope header in the following way:
+
+- `sentry-release` ➝ `release`
+- `sentry-environment` ➝ `environment`
+- `sentry-transaction` ➝ `transaction`
+- `sentry-userid` ➝ `user.id`
+- `sentry-usersegment` ➝ `user.segment`
+- `sentry-samplerate` ➝ `sample_rate`
+- `sentry-traceid` ➝ `trace_id`
+- `sentry-publickey` ➝ `public_key`
+
+## Unified Propagation Mechanism
+
+SDKs should follow these steps for any incoming and outgoing requests (in python pseudo-code for illustrative purposes):
+
+```python
+def collect_dynamic_sampling_context():
+  # Placeholder function that collects as many values for Dynamic Sampling Context
+  # as possible and returns a dict
+
+def on_incoming_request(request):
+  if has_header(request, "sentry-trace") and (not has_header(request, "baggage") or not has_sentry_value_in_baggage_header(request)):
+    # Request comes from an old SDK which doesn't support Dynamic Sampling Context yet
+    # --> we don't propagate baggage for this trace
+    transaction.baggage_locked = true
+    transaction.baggage = {}
+  elif has_header(request, "baggage") and has_sentry_value_in_baggage_header(request):
```

Comment:
How does `has_sentry_value_in_baggage_header` work?

Is there a specific list of list-items we need to be present for dynamic sampling to work?
I assume `samplerate` and `publickey` are required, are other fields as well?
Some of them are optional.

Reply:
> How does has_sentry_value_in_baggage_header work?

Added a pseudo implementation of the function to clarify: 0513094258f61cecaa3deddd390575e7d37a91b7 - We simply check if there is a key in the baggage header that starts with "sentry-" or not.

> Is there a specific list of list-items we need to be present for dynamic sampling to work?

There is no list yet but tried to clarify in 4a7508fdca382eb587fe7e2783257ed7fd064af4. I believe for DS we need `samplerate`, `publickey` and `traceid`. However I should note that right here we don't care at all about what values are actually needed. If something is missing in an incoming request, we propagate DSC as is, and don't try to add stuff. I again wanna put emphasis on the fact that DSC cannot be mutated (by the origin application or any application down the line) as soon it has been propagated.